frontend/catalyst/pennylane_extensions.py

```diff
@@ -2446,3 +2448,46 @@ def _get_batch_size(args_flat, axes_flat, axis_size):
         )
 
     return batch_size
+
+
+def callback(func):
+    """Decorator that will correctly pass the signature as arguments to the callback
+    implementation.
+
+    At the moment, this is a stub.
+    We can also use here isTracing to only bind with the callback while it is tracing,
+    otherwise, call the function itself.
+    """
+    signature = inspect.signature(func)
+    retty = signature.return_annotation
+
+    @wraps(func)
+    def bind_callback(*args, **kwargs):
+        callback_implementation(func, retty, *args, **kwargs)
+
+    return bind_callback
+
+
+def callback_implementation(
+    cb: Callable[..., Any], _result_shape_dtypes: Any, *args: Any, **kwargs: Any
+):
+    """TODO: Attribution. I looked into
+    https://jax.readthedocs.io/en/latest/_modules/jax/_src/callback.html#pure_callback"""
+    flat_args, in_tree = tree_flatten((args, kwargs))
```

Comment:
Putting a comment here so we don't forget before merging

Reply:
https://github.com/PennyLaneAI/catalyst/pull/540/commits/f6027fb039141fafebc640aacd43da6cee3c26e1 Anything I missed?